server/springboot/src/main/java/com/styra/tickethub_springboot/dao/model/Customer.java

```diff
@@ -0,0 +1,46 @@
+package com.styra.tickethub_springboot.dao.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotBlank;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import java.time.Instant;
+import java.util.Optional;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = Customer.TABLE_NAME)
+@JsonDeserialize(using = CustomerDeserializer.class)
+@JsonSerialize(using = CustomerSerializer.class)
```

Comment:
😵 Wow

Reply:
Yeah... by the time you count up all the annotations, I think there are more lines of code in this project that are just annotations than it would take to correctly implement the same thing in plain Java.

Reply:
Oh wow, that's a few annotations you've got there! :sweat_smile: